### PR TITLE
Add replyFmt and refactor format builder to support multi-arg formats

### PR DESCRIPTION
**Pull request by KnightNiwrem**
Discussion moved from: https://github.com/grammyjs/parse-mode/pull/4

### CODE REVIEW

File: src/format.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import type { MessageEntity } from './deps.deno.ts'
+
+export interface Stringable {
+  toString(): string;
+};
+
+class FormattedString {
```
Comment (KnorpelSenf) [inline src/format.ts:7]:
```suggestion
class FormattedString implements Stringable {
```

File: src/format.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+// Root format function
+const fmt = (rawStringParts: TemplateStringsArray, ...stringLikes: Stringable[]): FormattedString => {
+  let text = rawStringParts[0];
+  let entities: MessageEntity[] = [];
+
+  for (let i = 0; i < stringLikes.length; i++) {
+    const stringLike = stringLikes[i];
+    if (stringLike instanceof FormattedString) {
+      entities = entities.concat(stringLike.entities.map(e => ({ ...e, offset: e.offset + text.length })));
```
Comment (KnorpelSenf) [inline src/format.ts:62]: Why not `push`?

File: src/format.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    this.text = text
+    this.entities = entities
+  }
+
+  toString() {
+    return this.text;
+  }
+}
+
+const unwrap = (stringLike: Stringable): FormattedString => {
```
Comment (KnorpelSenf) [inline src/format.ts:21]: Is there a reason why you're avoiding functions so radically?

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
Looks good to me!

I only have a few nitpicks. Consider merging the `import` and `import type` statements into one. Also, these:

File: src/format.ts (hunk 3 above)
Comment (KnightNiwrem) [reply src/format.ts:21]: Not really. Consistency? Is there a reason to use function declaration - perhaps in situation where we need function declaration hoisting?

Comment (KnorpelSenf) [reply src/format.ts:21]: I just find it more readable. A function is a function, a const is a const. If a function needs to be assigned to a const, there's usually a good reason, hence the question. But it's a nit, this isn't blocking.

**Review by KnorpelSenf — APPROVED** [review]

**Pull request merged**